src/code_80718D8.c

```diff
@@ -151,3 +193,692 @@ void sub_8071D4C(Entity *pokemon, Entity *target, s32 exp)
+        strcat(buffer, *gUnknown_80F9B10);
+        r6++;
+    }
+    if(gFormatData_202DE30[3]> 0)
+    {
+        if(gUnknown_8107010[r6] != 0)
+            strcat(buffer, gUnknown_8107018[gUnknown_8107010[r6]]);
+        strcat(buffer, *gUnknown_80F9B34);
+        r6++;
+    }
+    if(gFormatData_202DE30[4]> 0)
+    {
+        if(gUnknown_8107010[r6] != 0)
+            strcat(buffer, gUnknown_8107018[gUnknown_8107010[r6]]);
+        strcat(buffer, *gUnknown_80F9B58);
+    }
+
+    if(buffer[0])
+        PrintFieldMessage(0, buffer, 1);
+
+    sub_806A2BC(GetLeader(), 0);
+}
+
+void LevelDownTarget(Entity *pokemon, Entity *target, u32 level)
+{
+    EntityInfo *info;
+    u8 flag;
+    s32 newLevel;
+    LevelData leveldata;
+    
+    if(EntityExists(target))
+    {
+        flag = FALSE;
+        info = target->info;
+        if(!IsClientOrTeamBase(info->joinedAt.joinedAt))
+        {
+            newLevel = info->level - level;
+            if(newLevel <= 0)
+                newLevel = 1;
+
+            if(newLevel != info->level)
+            {
+                GetPokemonLevelData(&leveldata, info->id, newLevel);
+                info->exp = leveldata.expRequired;
+                flag |= sub_80725A4(pokemon, target);
+            }
+        }
+        info->expGainedInTurn = 0;
+        info->unk149 = 0;
+
+        if(!flag)
+        {
+            SetMessageArgument(gAvailablePokemonNames, target, 0);
+            sub_80522F4(pokemon, target, *gUnknown_80F9B94);
+        }
+    }
+}
+
+bool8 sub_80723D0(Entity *pokemon, Entity *target, u8 param_3, u8 param_4)
+{
+    EntityInfo *info;
+    LevelData leveldata;
+    bool8 flag = FALSE;
+    s32 id;
+    s32 level;
+    s32 exp;
+    
+    if(!EntityExists(target)) return FALSE;
+    info = target->info;
+    id = info->id;
+    exp = info->exp;
+
+
+    if(IsClientOrTeamBase(info->joinedAt.joinedAt))
+        return FALSE;
+
+    for(level = info->level + 1; level <= 100; level++)
+    {
+        GetPokemonLevelData(&leveldata, id, level);
+        if(leveldata.expRequired > exp) break;
+
+        info->level = level;
+        info->exp = exp;
+
+        if(param_3)
+        {  
+            if(!info->isNotTeamMember)
+            {
+                sub_806A2BC(target, 0);   
+                if(!flag)
+                    sub_8083D58();
+                gFormatData_202DE30[0] = level;
+
+                SetMessageArgument_2(gAvailablePokemonNames, info, 0);            
+                sub_80528F4(target, *gUnknown_80F9E80);
+            }
+            else
+            {
+                if(!flag)
+                    sub_8083D58();
+                gFormatData_202DE30[0] = level;
+                SetMessageArgument_2(gAvailablePokemonNames, info, 0); 
+                sub_805239C(target, *gUnknown_80F9E80);
+            }
+        }
+        flag = TRUE;
+
+        info->maxHPStat += leveldata.gainHP;
+        info->HP += leveldata.gainHP;
+
+        if(info->maxHPStat > 0x3E6)
```

Comment:
These hex values are equal to 998 and 999 in decimal. Since these numbers are directly visible to the player, using decimal would be easier to read here.